packages/orm/src/EntityManager.ts

```diff
@@ -1905,6 +1900,67 @@ export class EntityManager<C = unknown, Entity extends EntityW = EntityW, TX ext
       );
     }
   }
+
+  fork(): EntityManager<C, Entity, TX> {
+    const oldEm = this;
+    const pendingEntities = oldEm.entities.filter((e) => getInstanceData(e).pendingOperation !== "none");
+    if (pendingEntities.length > 0) {
+      fail("Cannot fork an EntityManager with pending changes");
+    }
+    // copy the context so that it's distinct between the two ems and so we can update any references from the old em
+    // to the new one later
+    const ctx: C = { ...oldEm.ctx };
+    const newEm = new EntityManager<C, Entity, TX>(ctx, { em: oldEm });
+    // hydrate (ie, create) each entity from the oldEm in the newEm
+    oldEm.#entitiesByTag.forEach((entities, tag) => {
+      // tags are always defined by the base constructor, so we use that as hydrate expects the base class as its argument
+      const baseCstr = getConstructorFromTag(tag);
+      // We copy the raw row data for each entity to avoid any potential conflict. Then we ensure the correct concrete
+      // __class key is set for hydrate to use to actually instantiate the entity.
+      const rows = entities.map((e) => {
+        const { row, data } = getInstanceData(e);
+        return { ...(row ?? data), __class: e.constructor.name };
```

Comment:
Kinda wonder if we should only set `__class` specifically when CTI is involved, just to avoid iirc a few unnecessary lines in `hydrate` that run when `__class` is detected.